packages/keycloak-user-management/src/components/forms/UserForm/utils.tsx

```diff
@@ -0,0 +1,123 @@
+import { notification } from 'antd';
+import { history } from '@onaio/connected-reducer-registry';
+import { KeycloakUser } from '@opensrp/store';
+import { Dictionary } from '@onaio/utils';
+import { Dispatch, SetStateAction } from 'react';
+import { KeycloakService } from '@opensrp/keycloak-service';
+import {
+  KEYCLOAK_URL_USERS,
+  URL_ADMIN,
+  KEYCLOAK_URL_REQUIRED_USER_ACTIONS,
+  ERROR_OCCURED,
+} from '../../../constants';
+
+/**
+ * Handle form submission
+ *
+ * @param {Dictionary} values - form values
+ * @param {string} accessToken - keycloak API access token
+ * @param {string} keycloakBaseURL - keycloak API base URL
+ * @param {KeycloakService} keycloakServiceClass - keycloak API service class
+ * @param {Function} setSubmitting - method to set submission status
+ * @param {string} userId - keycloak user id, required when editing a user
+ */
+export const submitForm = (
+  values: Partial<KeycloakUser>,
+  accessToken: string,
+  keycloakBaseURL: string,
+  keycloakServiceClass: typeof KeycloakService,
+  setSubmitting: (isSubmitting: boolean) => void,
+  userId?: string
+): void => {
+  setSubmitting(true);
+
+  if (userId) {
+    const serve = new keycloakServiceClass(
+      accessToken,
+      `${KEYCLOAK_URL_USERS}/${userId}`,
+      keycloakBaseURL
+    );
+    serve
+      .update(values)
+      .then(() => {
+        setSubmitting(false);
+        history.push(URL_ADMIN);
```

Comment:
@kelvin-muchiri is this the correct history? looks like it was imported from `'@onaio/connected-reducer-registry'`

Reply:
@p-netm This is what is being used everywhere else

Reply:
should we switch to using `useHistory` from react-router @p-netm

Reply:
I Would say yes, but I do not know why we are using the one from here '@onaio/connected-reducer-registry', if it was simply a dev tools mistake say after an auto-import, then I would say we should use the first hand history object from `react-router
`